packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -372,6 +425,13 @@ export interface TableAttributes {
    * @default - no table stream
    */
   readonly tableStreamArn?: string;
+
+  /**
+   * KMS encryption key, if this queue is server-side encrypted by a KMS key.
+   *
+   * @default - no key arn
+   */
+  readonly keyArn?: string;
```

Comment:
You should surface the `IKey` only (as `encryptionKey` - similar to how [`@aws-cdk/aws-s3.BucketAttributes` is set up](https://docs.aws.amazon.com/cdk/api/latest/typescript/api/aws-s3/bucketattributes.html#aws_s3_BucketAttributes).